### DISCUSSION TITLE
Shelly door/window 2 e wifi mesh

### DISCUSSION THREAD

**Discussion by RobyBS** [indomus/forum#1663] [HUB: Home Assistant]
ho uno dei sensori in oggetto nella porta del garage interrato ... 

in questo garage riesco a far arrivare internet tramite una powerline ( FRITZ!Box per la precisione ) ...

se il sensore lavora collegato alla router principale , tutto funziona perfettamente  ... come anche altri sensori della stessa marca che ho in casa ...

quando vado in garage, il cambio di stato aperto/chiuso , non viene più rilevato e memorizzato da HA ... dopo varie prove sono giunto alla conclusione che il sensore connesso alla powerline ( in una rete mesh quindi stesso nome rete wifi e stessa password del router principale , non riesce a colloquiare con HA ... 

se mi collego al sensore , tramite indirizzo IP statico , dal mio cellulare per esempio, tutto funziona perfettamente .. quindi il problema è tra il sensore sotto wifi powerline e l'istanza di HA ( hassio per la precisione ) ...

avete dei suggerimenti di impostazioni da inserire nella configurazione di door/window2 ? vi è mai capitato ? 

potrei eventualmente valutare un altro sensore per la stessa funzione ( sapere se la basculante è aperta o chiusa in un dato momento ) , avete qualche suggerimento per evitare di incappare nello stesso problema ( wifi mesh / HA ) ? 

premetto che in garage ho anche shellyEM e shelly1 ( sotto wifi mesh ) che funzionano perfettamente ...

grazie!

**Comment by Scialla** [comment]
Non ti so aiutare nello specifico ma, hai pensato di usarli tramite mqtt?
Io ho un sensore collegato a una wifi sotto un range di IP completamente diverso da quello principale, in pratica un extender con range di IP americano 10.10.x.x sotto la rete principale 192.168.x.x.
Il sensore collegato alla rete 10.10.x.x. via mqtt trasmette il suo stato tranquillamente alla rete principale 192.168.x.x

**Comment by RobyBS** [comment]
anche io lo utilizzo sotto mqtt .. imposto il server sulla interfaccia del dispositivo ... 
ieri ha funzionato una o due volte ... ha cambiato stato e poi è diventato "non disponibile" su HA ... sempre raggiungibile su indirizzo IP del dispositivo quando è "sveglio" ... 
a proposito, scusa se approfitto, come mai diventano non disponibili , anche se funzionanti ( cambiano stato su indirizzo IP correttamente ) ... ?

**Reply by snakuzzo** [reply]
perché via MQTT, se il device va in deep sleep e nella configurazione hai impostato l'availabiility topic, nel momento in cui il device si disconnette dalla rete, e quindi dal broker MQTT, il broker invia il [LWT](https://indomus.it/formazione/mqtt-cose-e-come-funziona-il-last-will-and-testament-lwt/) topic... ovvero comunica che il device non è disponibile.
Nel caso, dovresti togliere l'availabiliity topic dalla configurazione su HA.

Se non usi l'availability topic (come abbiamo indicato nel [nostro articolo](https://indomus.it/guide/integrare-sensori-shelly-door-windows-2-a-home-assistant-via-mqtt/), non dovresti avere il comportamento che descrivi. Se invece hai lo stesso quel comportamento, devi metterti in ascolto sulla coda del broker (se non sei pratico puoi usare MQTT Explorer) e vedere cosa passa...perché in tal caso, se diventa unavailable, vuol dire che "qualcuno/qualcosa" fa la publish di cui ti parlavo prima.

Altra cosa, non ho quel device, controlla nelle impostazioni dello shelly se c'è l'LWT...nel caso disabilitalo

**Reply by RobyBS** [reply]
eccomi ... scusa il ritardo  e soprattutto grazie per la risposta ... 

provo a togliere il mqtt ( lo usavo solo per avere risposte più immediate dal device , o così penso invece di passare dal cloud ) ... 
il device non dispone della funzione di LWT ... non uso l'availability topic impostato in configurazione HA ... e vediamo cosa succede 👍

**Reply by snakuzzo** [reply]
Se non usi availability topic, l'unico momento in cui dovrebbe risultarti `unavailable` è quando avvii Core. Dopo, al primo cambio stato del sensore, non dovresti avere più stato sconosciuto...fino al successivo riavvio di Core

**Reply by RobyBS** [reply]
potrebbe darsi che sia questo il mio caso ... allora ... 
approfondisco ulteriormente e vi aggiorno , per i posteri .. 

ultima info .. mi viene un dubbio .. leggendo la vostra guida di integrazione del sensore shelly door windows in HA tramite MQTT .. fate l'esempio di inserire ad uno ad uno tutti i sensori ( battery, state, tilt, vibration, etc ) in configurazione... 
a me succede che usando l'integrazione ufficiale Shelly di HA , questi "sensori" ( binary_sensor )  mi vengono creati automaticamente e li trovo disponibili in HA ... 
sbaglio qualcosa o devo comunque "dichiararli" in configurazione ? ... 
grazie

**Reply by snakuzzo** [reply]
No...è così.
Non capisco cos'è che non ti torna

**Reply by RobyBS** [reply]
visto che i sensori me li crea automaticamente l'applicazione SHELLY , non è necessario dichiararli in configuration.yaml come indicato nella Vs. guida ... giusto ?

**Reply by snakuzzo** [reply]
esatto.
Usa l'integrazione nativa ed elimina tutte le entità MQTT che hai creato.

**Reply by RobyBS** [reply]
perfetto .. e grazie per aver "dipanato" il dubbio che mi è venuto ... 

se riavvio HA ... i sensori che prima erano "unavailable" diventano "chiusi " ... 
ma forse è solo perche HA legge l'ultimo stato noto ricevuto ... mmmm....

**Reply by RobyBS** [reply]
con un sensore ho risolto .. tutto perfetto ! ... che tra l'altro era quello sotto rete mesh ... 

con un altro invece continua a non essere disponibile ... 
sembrerebbe un problema di batterie ... ovvero basterebbe al sensore shelly D/W 2 arrivare al di sotto del 90% di batteria a renderlo "non più disponibile" .. il voltaggio calerebbe drasticamente al calare del solo 10/15% di batteria ... e va beh ... faccio anche questa prova e vedo  di aggiornarvi ... 
più che altro per i posteri che dovessero incappare nello stesso problema ... 👍
